### ISSUE TITLE
Admin Interface Drag and Drop Interactions Unclear

### ISSUE THREAD

**Issue by Santoz2** [ManifoldScholar/manifold#2437]
For content creation in Manifold, Drag and Drop is an important feature. This may be seen in the Layout page for a book. While a keyboard user can interact with these drag and drop elements, there are issues with the interaction for Screen Reader Users

## Expected Behavior
Users should be aware of how to interact with elements, as well as how their interactions affect those elements. For Drag and Drop specifically, a user should be told how to interact with the element. Further, the user should be given status updates, such as if an element is being used, and where it is being moved to.

Hadi recommends adding documentation for this feature, as well as a link to that documentation, so that a reader may understand how it works.

## Current Behavior
A keyboard user may interact with Manifold's Drag and Drops, but it is not initially clear how to do so. A Screen Reader User is told that they are dragging something but not told where it is being dragged to/what elements are moving.

## Steps to Reproduce
1. Navigate to Admin Interface (via Enter Admin Mode)
2. Navigate to an editable project
3. Navigate to the Layout page
4. Navigate to the Layout subsection in the Content Blocks section
5. Navigate to the "Button Draggable" button on one of the Layout elements
6. Note how there are no instructions for how to use this element
7. Press Space and notice how the user is told that they are dragging the element
8. Press up or down arrow keys. Note how the element successfully moves, but a Screen Reader user is told nothing of this interaction.
9. Press space again, and notice how the user is not told that the action has been completed.

## Testing Environment
NVDA 2019.2.1
ESR Firefox 68.2.0 esr

**Comment by zdavis** [comment]
Duplicate of #2417. This issue has more detail, so we'll close #2417.

**Comment by dananjohnson** [comment]
This issue would be resolved by bumping `react-beautiful-dnd` from `4.0.1` to at least `5`. (`12.2` is current.) Releases beginning with [`5`](https://github.com/atlassian/react-beautiful-dnd/releases/tag/v5.0.0) provide instructions to screen readers telling them how the user can interact with the interface, and announcing user actions. There's a [handy readme](https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/guides/screen-reader.md) in the repo that discusses everything.

**Comment by zdavis** [comment]
@bot bug

**Comment by zdavis** [comment]
Thank you for taking the time to report this bug. We've reviewed the issue and agree that this should be fixed. We’re adding a “planned” label to indicate that we consider this bug as part of our current work plan.

_This was an automated message, but please don't hesitate to reply. Our team watches these issues closely and will respond as soon as we're able to!_

**Comment by Santoz2** [comment]
Accessibility Triage Notes: UW's ATX team considers this to be a priority 1 issue. It is both unclear how to interact with this, and there is no feedback during the interaction. That said, if instructional text were added, it may be considered a priority 2 issue, because a user could complete their task, though it would still be confusing.

**Comment by Santoz2** [comment]
While running through this issue again for the Accessibility Triage, I ran into this variant of the drag and drop feature:

![image](https://user-images.githubusercontent.com/43553368/75202004-d86ed900-571e-11ea-98aa-307433dcbdb1.png)

Here, it is possible to drag and drop within a category, but not between them (for example, here, a keyboard user can drag an item around in the uncategorized secion, but cannot drag it to the test category, though a mouse user can)

**Comment by dananjohnson** [comment]
Re: @Santoz2's comment, we are limited here by how `react-beautiful-dnd` handles multi-column keyboard navigation. In this case, <kbd>↑</kbd> and <kbd>↓</kbd> move `<Draggable />` items up and down in the `<Droppable />`. <kbd>→</kbd> and <kbd>←</kbd> are then reserved for moving items between the two categories. However, this isn't possible when the two categories are stacked vertically.

We have a similar issue with the CTA blocks in the Hero builder, where items can be moved between left and right slots, but not the ones above or below. (<kbd>↑</kbd> and <kbd>↓</kbd> just move items within a given slot.)

![Kapture 2020-03-06 at 14 43 12](https://user-images.githubusercontent.com/12454203/76128371-04107f80-5fb9-11ea-93da-d343fff7876c.gif)

The problem here is the grid layout. We'd need all four slots in a single row to make full keyboard navigation possible.

![Kapture 2020-03-06 at 14 57 21](https://user-images.githubusercontent.com/12454203/76129007-e3492980-5fba-11ea-91c5-6be10b47767c.gif)

There's an [open issue](https://github.com/atlassian/react-beautiful-dnd/issues/316#issuecomment-364794806) in the `rbd` repository to allow grid reordering, but it's not a planned feature at the moment.

So in short, multiple vertical lists have to be arranged laterally, and multiple horizontal lists have to be arranged vertically. (See [`rbd` documentation](https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/sensors/keyboard.md) for more info.)

Do we have alternatives to rethinking the design? `react-beautiful-dnd` has very strong accessibility features otherwise, all of which would be time-consuming to recreate if we opted for a more custom solution.

**Comment by zdavis** [comment]
Hey, @dananjohnson can't we just attach our own event listener to ← and → — and when it's triggered, we can just update state to move the item to the right container? We might lose the animation, but it seems like it would address the accessibility concern.

**Comment by dananjohnson** [comment]
That sounds like it could work. I'll investigate on Monday.

**Comment by Santoz2** [comment]
4/27/2020 UW ATX Verification: The issue about Drag and Drop Notifications is mostly fixed; there is now instructional text for Screen Readers on how to interact with elements, as well as text during the interaction.

However, there is an issue with the interaction text;

![image](https://user-images.githubusercontent.com/43553368/80418286-08a43680-888c-11ea-8abe-22437d9c3975.png)

The statement for moving items runs together as one word, which makes the interaction a bit unclear.

Once this is fixed, I think the text is very descriptive and clear!

Tested with Firefox + NVDA

**Comment by Santoz2** [comment]
> Tested with Firefox + NVDA

Apologies for missing this, but this actually functions differently on Chrome, Firefox, and Edge:

- On Chrome, this reads perfectly fine.
- On Firefox, the above behavior is seen, with every word running together.
- On Edge, there is still no announcement for dragging and dropping, but the instructions are announced.
